### PR TITLE
HOLD ON RELEASE: 46758: Change Password in Admin Console

### PR DESCRIPTION
**Pull request by jonquil2002**
SC: https://app.shortcut.com/replicated/story/46758/blue-team-change-password-in-admin-console-ui

Preview: https://deploy-preview-326--replicated-docs.netlify.app/enterprise/auth-changing-passwords

### CODE REVIEW

File: docs/enterprise/auth-changing-passwords.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+When you install an application for the first time, the Replicated admin console is secured with a single shared password for all users. We recommend that you change your password to a unique password for security.
+
+:::note
+Users with Identity Provider (IDP) access cannot change their password using this procedure. If an attempt is made, IDP users receive a message in the user interface to contact the identity service provider to change it. For more information about resetting an IDP user password, see [Resetting Authentication](auth-identity-provider#resetting-authentication) in _Using an Identity Provider for User Access (Beta)_.
+:::
+
+To change a password:
+
+1. Log in to the admin console using the your current password.
```
Comment (alicenstar) [inline docs/enterprise/auth-changing-passwords.md:11]: "Log in to the admin console using **the your** current password." - little typo here.

File: docs/enterprise/auth-changing-passwords.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+1. Click **Change Password** in the lower right corner of the dashboard.
```
Comment (alicenstar) [inline docs/enterprise/auth-changing-passwords.md:12]: I don't know if saying "dashboard" is too specific. You can change your password on any page, not just the dashboard.

File: docs/enterprise/auth-changing-passwords.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+1. In the Change Admin Console Password dialog, edit the fields.
+
+    - The new password must be at least 6 characters and should not be the same as your current password.
```
Comment (alicenstar) [inline docs/enterprise/auth-changing-passwords.md:15]: Maybe add a note about how the user must also provide the current password.

Comment (alicenstar) [reply docs/enterprise/auth-changing-passwords.md:15]: Nevermind, I see you mention it later.

File: docs/enterprise/auth-changing-passwords.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    - The New Password and Confirm New Password fields must match each other.
+    - Click **Show** or **Hide** on each field to view or hide the password.
+
+1. Click **Change Password**.
+
+  If the current password is missing or incorrect, or if the new password fields do not meet the criteria in the previous step, an error message states the specific problem.
```
Comment (alicenstar) [inline docs/enterprise/auth-changing-passwords.md:21]: nitpick (sorry these are all nitpicks): I feel like the end wording is a little awkward. Maybe something like "an error message will show stating the specific problem."

File: docs/enterprise/auth-changing-passwords.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  When the password change succeeds, the current session closes and the Log In page opens.
```
Comment (alicenstar) [inline docs/enterprise/auth-changing-passwords.md:23]: Maybe change to "and you will be redirected to the Log In page".

File: docs/enterprise/auth-changing-passwords.md (hunk 2 above)
Comment (JustinN-Replicated) [reply docs/enterprise/auth-changing-passwords.md:12]: 'app manager console.' may be more accurate

Comment (JustinN-Replicated) [reply docs/enterprise/auth-changing-passwords.md:12]: or 'admin console.' sorry - didn't notice the usage of this above (vs app manager)

File: docs/enterprise/auth-changing-passwords.md (hunk 4 above)
Comment (JustinN-Replicated) [reply docs/enterprise/auth-changing-passwords.md:21]: I agree, I don't think we need to give every failure scenario (too wordy, I think it is OK that people expect certain things since password forms are somewhat familiar). 

Suggest replacing with something like:
"If there are any issues with changing the password, an error message will display with the problem so it can be fixed."

**Review by JustinN-Replicated — COMMENTED** [review]
Once comments are addressed let me know and I will approve; I also made a change as the new password _**must not**_ be the same as the old password (so 'should' was a bit inaccurate)

File: docs/enterprise/auth-changing-passwords.md (hunk 1 above)
Comment (jonquil2002) [reply docs/enterprise/auth-changing-passwords.md:11]: TY!

File: docs/enterprise/auth-changing-passwords.md (hunk 3 above)
Comment (jonquil2002) [reply docs/enterprise/auth-changing-passwords.md]: Yes, and we don't describe fields that are self-evident - unless there is a limitation they need to know about.

File: docs/enterprise/auth-changing-passwords.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Changing an Admin Console Password
+
+When you install an application for the first time, the Replicated admin console is secured with a single shared password for all users. We recommend that you change your password to a unique password for security.
+
+:::note
+Users with Identity Provider (IDP) access cannot change their password using this procedure. If an attempt is made, IDP users receive a message in the user interface to contact the identity service provider to change it. For more information about resetting an IDP user password, see [Resetting Authentication](auth-identity-provider#resetting-authentication) in _Using an Identity Provider for User Access (Beta)_.
```
Comment (paigecalvert) [inline docs/enterprise/auth-changing-passwords.md:6]: Suggestion to clarify use of "it":

"to contact the identity service provider to change **it**"
-->
"to contact the identity service provider to change **their password**"

File: docs/enterprise/auth-changing-passwords.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Changing an Admin Console Password
+
+When you install an application for the first time, the Replicated admin console is secured with a single shared password for all users. We recommend that you change your password to a unique password for security.
```
Comment (JustinN-Replicated) [inline docs/enterprise/auth-changing-passwords.md:3]: This is more so really when KOTS is installed via kURL (IIRC when you install kots in an existing cluster, you provide a password and it is masked when it is entered). @pavansokkenagaraj does that sound correct? Just want to make sure we are not completely misleading here (IE uploading a license to install an app or to reset a password when it wasn't displayed like how kURL does isn't the right way to guide users - we really want to guide them to change either 1) when they desire /want to and 2) encourage them to change after a kURL install like we did on the 'sister ticket' to this one). 

WDYAT?

File: docs/enterprise/auth-changing-passwords.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+:::note
+Users with Identity Provider (IDP) access cannot change their password using this procedure. If an attempt is made, IDP users receive a message in the user interface to contact the identity service provider to change it. For more information about resetting an IDP user password, see [Resetting Authentication](auth-identity-provider#resetting-authentication) in _Using an Identity Provider for User Access (Beta)_.
+:::
+
+To change a password:
```
Comment (paigecalvert) [inline docs/enterprise/auth-changing-passwords.md:9]: Suggestion:
"To change your admin console password:"

File: docs/enterprise/auth-changing-passwords.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+1. Log in to the admin console using your current password.
+1. Click **Change Password** in the lower right corner of any page.
+1. In the Change Admin Console Password dialog, edit the fields.
```
Comment (paigecalvert) [inline docs/enterprise/auth-changing-passwords.md:13]: The three items in the bulleted list don't feel like they fit neatly together as a single list. The first two are requirements for the password, and the last is more like a tip.

Suggestion: The requirements could be grouped together in a list, making it clear that they are requirements for the password:

In the Change Admin Console Password dialog, edit the fields to meet the following requirements:
- The new password must be...
- The New Password and Confirm ...

"Click show or hide" is not a requirement, so I was thinking this could be handled by:
- Removing it from this procedure all together (might not be necessary to call out the show/hide feature of the UI in the docs)
- Moving it to a note or a separate line nested under the step
- Moving it to a separate (Optional) step in the procedure

**Review by paigecalvert — APPROVED** [review]
Thank you @jonquil2002 ! Looks great

File: docs/enterprise/auth-changing-passwords.md (hunk 7 above)
Comment (jonquil2002) [reply docs/enterprise/auth-changing-passwords.md:3]: @JustinN-Replicated @pavansokkenagaraj That first sentence is also in the Identity Access topic: https://docs.replicated.com/enterprise/auth-identity-provider

So to clarify, the single shared password only happens when it's a Kubernetes installer (kURL) cluster?

Comment (JustinN-Replicated) [reply docs/enterprise/auth-changing-passwords.md:3]: The identiy access beta is only applicable to kURL installs (it is not supported on existing cluster yet). 

The single shared password is used for all of kots if the above beta is not turned on; the password is displayed when kots is installed for the first time via kURL (AFAIK all other permutations it is not displayed in raw text). @sgalsaleh I think we had discussed this on slack but wanted to loop you in to keep me honest ;)

Comment (pavansokkenagaraj) [reply docs/enterprise/auth-changing-passwords.md:3]: > when you install kots in an existing cluster, you provide a password and it is masked when it is entered

Yes

**Comment by JustinN-Replicated** [comment]
@paigecalvert @jonquil2002 @alicenstar  @pavansokkenagaraj 
Just wanted to loop everyone in as I made a change related to my last comment. At this point I think this one should be good to go.

**Review by JustinN-Replicated — APPROVED** [review]
lgtm
